pkg/google/natgateway/natgateway.go

```diff
@@ -0,0 +1,178 @@
+			Name: prometheus.BuildFQName(cloudcost_exporter.ExporterName, subsystem, "next_scrape"),
+			Help: "The next time the exporter will scrape GCP billing data for NAT Gateway",
+		}),
+	}
+}
+
+type Config struct {
+	ProjectId      string
+	Projects       string // comma-separated list
+	ScrapeInterval time.Duration
+}
+
+type Collector struct {
+	config             *Config
+	cloudCatalogClient *billingv1.CloudCatalogClient
+	ctx                context.Context
+	nextScrape         time.Time
+	metrics            *Metrics
+}
+
+func New(cfg *Config, cloudCatalogClient *billingv1.CloudCatalogClient) (*Collector, error) {
+	ctx := context.Background()
+	c := &Collector{
+		config:             cfg,
+		cloudCatalogClient: cloudCatalogClient,
+		ctx:                ctx,
+		nextScrape:         time.Now().Add(-cfg.ScrapeInterval),
+		metrics:            NewMetrics(),
+	}
+	return c, nil
+}
+
+func (c *Collector) Name() string { return "NATGATEWAY" }
+
+func (c *Collector) Register(registry provider.Registry) error {
+	registry.MustRegister(c.metrics.HourlyRateGauge)
+	registry.MustRegister(c.metrics.DataProcessingGauge)
+	registry.MustRegister(c.metrics.NextScrapeGauge)
+	return nil
+}
+
+func (c *Collector) Describe(ch chan<- *prometheus.Desc) error { return nil }
+
+func (c *Collector) Collect(ch chan<- prometheus.Metric) error {
+	c.collect()
+	return nil
+}
+
+// CollectMetrics exists for consistency with other collectors; returns 1 to indicate it ran.
+func (c *Collector) CollectMetrics(ch chan<- prometheus.Metric) float64 {
+	c.collect()
+	return 1
+}
+
+func (c *Collector) collect() {
+	now := time.Now()
+	if c.nextScrape.After(now) {
+		return
+	}
+	c.nextScrape = time.Now().Add(c.config.ScrapeInterval)
+	c.metrics.NextScrapeGauge.Set(float64(c.nextScrape.Unix()))
+
+	serviceName, err := common.GetServiceName(c.ctx, c.cloudCatalogClient, "Networking")
+	if err != nil {
+		log.Printf("gcp natgateway: failed to get Compute Engine service: %v", err)
+		return
+	}
+
+	skus := common.GetPricing(c.ctx, c.cloudCatalogClient, serviceName)
+
+	projects := common.ParseProjects(c.config.ProjectId, c.config.Projects)
+
+	// Reset metrics for a fresh scrape
+	c.metrics.HourlyRateGauge.Reset()
+	c.metrics.DataProcessingGauge.Reset()
+
+	for _, sku := range skus {
+		if sku == nil {
+			continue
+		}
+		desc := strings.ToLower(sku.Description)
+		if !isCloudNATSKU(desc) {
+			continue
+		}
+
+		priceUSD := firstTierPriceUSD(sku)
+		if priceUSD == 0 {
+			continue
+		}
+
+		// service regions are like: [us-central1, europe-west1, ...]
+		for _, region := range sku.ServiceRegions {
+			for _, project := range projects {
+				skuID := sku.GetName()
+				skuName := sku.Description
+				if isDataProcessing(desc) {
+					c.metrics.DataProcessingGauge.WithLabelValues(region, project, skuID, skuName).Set(priceUSD)
+				} else {
+					c.metrics.HourlyRateGauge.WithLabelValues(region, project, skuID, skuName).Set(priceUSD)
+				}
```

Comment:
We're including `Networking Cloud NAT IP Usage` here, which is a [different charge type](https://cloud.google.com/nat/pricing):
> Price per hour for a static or an ephemeral external IP address used by a NAT gateway**
> $0.005

Reply:
I see that this is specified in the metric via the label `sku_name` - perfect!

Reply:
BTW removed google support, moving it to a followup PR.